plugins/helper/api_collector.go

```
@@ -335,6 +337,13 @@ func (collector *ApiCollector) fetchAsync(reqData *RequestData, handler func(int
 			panic(err)
 		}
 	}
+	var reqBody interface{}
+	if collector.args.RequestBody != nil {
+		reqBody = collector.args.RequestBody(reqData)
+		if err != nil {
```

should we panic here? Why not just return the error?